site/src/analytics-integrations.ts

```diff
@@ -12,77 +11,114 @@ import {
 
   @internalProperty() private _currentTableSize = 30;
   @internalProperty() private _currentTablePage = 0;
 
   protected firstUpdated(_changedProperties: PropertyValues) {
     super.firstUpdated(_changedProperties);
 
-    const dataKeys = Object.keys(this.data!);
-    const lastEntry = this.data![dataKeys[dataKeys.length - 1]];
-    this._integrations = Object.keys(lastEntry.integrations).map(
-      (integration) => {
-        return {
-          integration,
-          installations: lastEntry.integrations[integration],
-        };
-      }
-    );
+    this.getData();
   }
 
   render() {
     if (this._integrations === undefined) {
       return html``;
     }
 
-    const tableStart = this._currentTablePage * this._currentTableSize;
-    const tableEnd =
-      tableStart + this._currentTableSize <= this._integrations.length
-        ? tableStart + this._currentTableSize
-        : this._integrations.length;
+    let idx = 0;
 
-    const tableData = this._integrations
+    const sortedTableData = this._integrations
       .sort(
         (a, b) =>
-          b.installations - a.installations ||
-          a.integration.localeCompare(b.integration)
+          b.installations - a.installations || a.title.localeCompare(b.title)
       )
-      .slice(tableStart, tableEnd);
+      .map((entry) => {
+        idx++;
+        return { ...entry, idx };
+      })
+      .filter((entry) =>
+        this._filter
+          ? entry.title.toLowerCase().includes(this._filter.toLowerCase()) ||
+            entry.domain.includes(this._filter.toLowerCase())
+          : true
+      );
+
+    const tableStart = this._currentTablePage * this._currentTableSize;
+    const tableEnd =
+      tableStart + this._currentTableSize <= sortedTableData.length
+        ? tableStart + this._currentTableSize
+        : sortedTableData.length;
 
-    console.log(tableEnd);
+    const tableData = sortedTableData.slice(tableStart, tableEnd);
 
     return html`
-      <h3>Integration usage</h3>
+      <div class="header">
+        <h3>Integration usage</h3>
+        ${!isMobile
+          ? html` <div class="search">
+              <input
+                class="searchbar"
+                .value=${this._filter}
+                @input=${this._filterChange}
+                placeholder="Search"
+              />
+              ${this._filter
+                ? html` <mwc-icon-button
+                    class="clear-search"
+                    @click=${this._clearFilter}
+                  >
+                    <svg>
+                      <path d=${mdiClose} />
+                    </svg>
+                  </mwc-icon-button>`
+                : ""}
+            </div>`
+          : ""}
+      </div>
+
       <table>
         <tr class="table-header">
+          ${!isMobile ? html`<th></th>` : ""}
           <th>Integration</th>
           <th>Installations</th>
         </tr>
         ${tableData.map(
           (entry) => html`
```

Comment:
```suggestion
          (entry, idx) => html`
```

Reply:
Did that in L55 to keep placement when using search

Reply:
Ah ok :-)